From 6a70eb89593abf614828a63c5f40c26c262c595b Mon Sep 17 00:00:00 2001
From: Andreas Kempf <aakempf@gmail.com>
Date: Fri, 31 Jul 2015 10:20:25 +0200
Subject: [PATCH] hwdb: Add mouse DPI info for Logitech G402.

DPI information was taken from the device's setup guide.

Additional (shortened) information from udevadm info:

P: .../usb2/2-1/2-1:1.0/0003:046D:C07E.0002/input/input6/event2
N: input/event2
S: input/by-id/usb-Logitech_Gaming_Mouse_G402_6D93117D5049-event-mouse
S: input/by-path/pci-0000:00:12.0-usb-0:1:1.0-event-mouse
...
E: ID_BUS=usb
E: ID_INPUT=1
E: ID_INPUT_MOUSE=1
E: ID_MODEL=Gaming_Mouse_G402
E: ID_MODEL_ENC=Gaming\x20Mouse\x20G402
E: ID_MODEL_ID=c07e
E: ID_PATH=pci-0000:00:12.0-usb-0:1:1.0
E: ID_PATH_TAG=pci-0000_00_12_0-usb-0_1_1_0
E: ID_REVISION=9002
E: ID_SERIAL=Logitech_Gaming_Mouse_G402_6D93117D5049
E: ID_SERIAL_SHORT=6D93117D5049
E: ID_TYPE=hid
E: ID_USB_DRIVER=usbhid
E: ID_USB_INTERFACES=:030102:030000:
E: ID_USB_INTERFACE_NUM=00
E: ID_VENDOR=Logitech
E: ID_VENDOR_ENC=Logitech
E: ID_VENDOR_ID=046d
E: LIBINPUT_DEVICE_GROUP=3/46d/c07e/111:usb-0000:00:12.0-1
E: MAJOR=13
E: MINOR=66
E: SUBSYSTEM=input
...

(cherry picked from commit 3407fcd575b31416ae1658572590d0b4973f5ace)
---
 hwdb/70-mouse.hwdb | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hwdb/70-mouse.hwdb b/hwdb/70-mouse.hwdb
index 4c84f0a897..68e849b3d8 100644
--- a/hwdb/70-mouse.hwdb
+++ b/hwdb/70-mouse.hwdb
@@ -239,6 +239,10 @@ mouse:usb:v046dpc245:name:Logitech Gaming Mouse G400:
 mouse:usb:v046dpc24c:name:Logitech G400s Optical Gaming Mouse:
  MOUSE_DPI=400@1000 *800@1000 2000@1000 4000@1000
 
+# Logitech G402 Hyperion Fury
+mouse:usb:v046dpc07e:name:Logitech Gaming Mouse G402:
+ MOUSE_DPI=400@1000 *800@1000 1600@1000 3200@1000
+
 # Logitech B605 Wireless Mouse (also M505)
 mouse:usb:v046dp101d:name:Logitech B605:
 mouse:usb:v046dp101d:name:Logitech M505:
